src/components/Integrations/Table/IntegrationTest.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { Button, Modal, TextInput } from '@patternfly/react-core';
+import { useNotification } from '../../../utils/AlertUtils';
+
+const IntegrationTestModal = ({ integrationUUID, isModalOpen, onClose }) => {
+  const [inputValue, setInputValue] = useState('');
+  const { addSuccessNotification, addWarningNotification } = useNotification();
+
+  const placeholderText =
+    'Congratulations! The integration you created on https://console.redhat.com was successfully tested!';
+
+  const failedTestText =
+    'Your integration test has unfortunately failed, please verify your integration information.';
+
+  const handleNotificationTest = async (notificationMessage: string) => {
+    const endpointURL = `/api/integrations/v1/endpoints/${integrationUUID}/test`;
+
+    const body = { message: notificationMessage };
+
+    try {
+      const response = await axios.post(endpointURL, body);
+      response?.status === 204
+        ? addSuccessNotification('Integration Test', notificationMessage)
+        : addWarningNotification(
+            'Failed Test',
+            `Error through server response: ${response.status}`
+          );
+      onClose();
```

Comment:
@arivepr since `onClose();` is executed in both the `try` and `catch`, should we move it to `finally` instead and dedupe? Also I'd imagine this conditional logic is hit when the server returns 400 or 500 level errors. Is showing a notification with just the `response.status` actually enough information?